pin/pin.go

```diff
@@ -221,30 +231,51 @@ func (p *pinner) RemovePinWithMode(key util.Key, mode PinMode) {
 func LoadPinner(d ds.ThreadSafeDatastore, dserv mdag.DAGService) (Pinner, error) {
 	p := new(pinner)
 
+	rootKeyI, err := d.Get(pinDatastoreKey)
```

Comment:
ah, well, if `/local/pins` and `/local/pins/direct` are both ipfs objects, and `/local/pins` points to `/local/pins/direct`, then we don't need to store the object at `/local/pins/direct` at at that key.

my hope was to have all the new pinning objects stored as usual (under their hashes). And yeah, it makes sense to do it as a separate commit.

But migrations are expensive (in risk). every time we add one we run the risk of breaking something for the user, and leaving them in a weird state. I don't think we should ship this to users only to have another migration pushed on them right after. (and records are coming right around the corner)

At the same time i dont want to hold up merging this. Maybe we can have a branch called `object-repo` or something and merge this (and S3 stuff) into that. We can merge it into master once we make the repo an object?

@whyrusleeping @tv42 thoughts?